scripts/setup.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+# Get tools
+asdf install
+asdf reshim
+
+# Setup Elixir
+mix deps.get
+
+# Setup JS
+cd assets && npm install && cd --
```

Comment:
Why the `cd --` here?  Did you mean `cd ..`?  If so, you could also do `(cd assets && npm install)`.

Reply:
`cd --` (switch to directory before last cd) accomplishes the same things as parenthesizing it--returning to the root directory after the operation.

I started preferring the idiom at some point, because it doesn't fork a subshell like parens do. Not chosen with any particular intent, just a dim preference on my part.

Reply:
Are you sure?  My understanding of the `--` was that it was a built-in bash option that causes the shell to not perform any further option parsing, important for some cases like where the name of the argument begins with a hyphen.  Since there are no arguments to the right of the `--`, I read that as the same as `cd`, which takes you back to your home directory.  Running that line on a bash prompt on my system also takes me to my home directory.

My understanding on why the parens version is sometimes preferred is because it always take you back to the directory you were in at the time you entered the parens, regardless of whether the previous command succeeded or not.  In the version you currently have, without the `set -e` to exit the script, it would try to run the following commands from within the assets directory if the  the `npm install` step failed.  Either variant is fine though with the `set -e`.

Reply:
Shoot, you're right--I confused this with `git checkout --` which does what I described, but for branches.